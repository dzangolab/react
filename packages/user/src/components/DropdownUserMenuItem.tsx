import React from "react";
import { Link } from "react-router-dom";

interface Properties {
  children: React.ReactNode;
  icon?: string;
  route?: string;
  showIcon?: boolean;
}

const DropdownUserMenuItem = ({
  children,
  route,
  icon,
  showIcon,
}: Properties) => {
  if (route) {
    return (
      <Link to={route}>
        {showIcon && <i className={icon}></i>}
        {children}
      </Link>
    );
  }

  return (
    <>
      {showIcon && <i className={icon}></i>}
      {children}
    </>
  );
};

export default DropdownUserMenuItem;
