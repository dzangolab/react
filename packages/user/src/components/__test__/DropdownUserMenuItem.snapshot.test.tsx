import React from "react";
import { create } from "react-test-renderer";
import { expect, test } from "vitest";

import DropdownUserMenuItem from "../DropdownUserMenuItem";

import type {
  ReactTestRenderer,
  ReactTestRendererJSON,
} from "react-test-renderer";

function toJson(component: ReactTestRenderer) {
  const result = component.toJSON();

  expect(result).toBeDefined();
  expect(result).not.toBeInstanceOf(Array);

  return result as ReactTestRendererJSON;
}

test("Component matches snapshot", () => {
  const component = create(<DropdownUserMenuItem>test</DropdownUserMenuItem>);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
