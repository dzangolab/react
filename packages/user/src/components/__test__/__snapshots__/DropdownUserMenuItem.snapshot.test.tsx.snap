// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Component matches snapshot 1`] = `
<div>
  test
</div>
`;
